### PR TITLE
System timer

### PR DESCRIPTION
**Pull request by aqib-ublox**
### Pull request type

- [ ] Fix
- [ ] Refactor
- [ ] New target
- [-] Feature
- [ ] Breaking change

### CODE REVIEW

File: targets/TARGET_ublox/TARGET_R5030/us_ticker.c [hunk 1]
```diff
@@ -133,49 +147,58 @@ void IRQ1_TMR0_Handler(void)
...
     core_util_critical_section_enter();
 
-     Get the timer value, adding the offset in case we've been moved
-     * around by user activity, inverting it (as a count-up timer is
-     * expected), then scaling it to useconds and finally adding the
-     * usecond overflow value to make up the 32-bit usecond total 
-    timeValue = divide_by_48(~(TIMER0_TIME + g_user_interrupt_offset)) + g_us_overflow;
-
-     Put interrupts back 
+    /* If TIMER2 is overflowed (value > load register) value so it means
```
Comment (RobMeades) [inline targets/TARGET_ublox/TARGET_R5030/us_ticker.c]: Nit pick: worth correcting spacing/spelling here.

**Review by RobMeades — COMMENTED** [review]
Aside from the one nit-pick this is fine by me assuming that the us_ticker platform tests pass.  One question: is this implementation to the very latest us_ticker API?  I ask because I think in the most recent API the init function is required to be called before a read is performed, so we wouldn't need the check that you do here.

File: targets/TARGET_ublox/TARGET_R5030/us_ticker.c (hunk 1 above)
Comment (aqib-ublox) [reply targets/TARGET_ublox/TARGET_R5030/us_ticker.c]: Yeah @RobMeades you are right this is not new us-ticker API implementation, it is Ticker class implementation for KM which was fixed some time ago in system_timer branch but that was not merged to master so as now i am going to update us_ticker new implementation for C027 i realize that master branch dont have chnage set committed by me so this was done in this way.

Comment (RobMeades) [reply targets/TARGET_ublox/TARGET_R5030/us_ticker.c]: Understood, that's fine, we can do it incrementally.

**Comment by aqib-ublox** [comment]
@RobMeades formatting updated

**Review by RobMeades — APPROVED** [review]
LGTM.

**Pull request merged**